### DISCUSSION TITLE
How to convert between RGB colourspaces with different illuminants?

### DISCUSSION THREAD

**Discussion by hminle** [colour-science/colour#832] [General]
Hi Colour Team,

I am quite confused about how to convert between RGB colourspaces with different illuminants. For example, sRGB with D65 to ProPhotoRGB with D50.

Here is the example in Colour Documentation:
```python
from colour.models import (RGB_COLOURSPACE_sRGB, RGB_COLOURSPACE_PROPHOTO_RGB)
RGB = np.array([0.45595571, 0.03039702, 0.04087245])
RGB_to_RGB(RGB, RGB_COLOURSPACE_sRGB, RGB_COLOURSPACE_PROPHOTO_RGB)
```

But I think it should be like this:

```python
from colour.models import (RGB_COLOURSPACE_sRGB, RGB_COLOURSPACE_PROPHOTO_RGB)
ILLUMINANT_D50 = colour.CCS_ILLUMINANTS['CIE 1931 2 Degree Standard Observer']['D50']
srgb_colourspace_D50 = RGB_COLOURSPACE_sRGB.chromatically_adapt(ILLUMINANT_D50)
RGB = np.array([0.45595571, 0.03039702, 0.04087245])
RGB_to_RGB(RGB, srgb_colourspace_D50, RGB_COLOURSPACE_PROPHOTO_RGB)
```

And `chromatic_adaptation_transform='CAT02'` by default, what if we specify `chromatic_adaptation_transform=None`?

Thanks.

**Comment by thomasmansencal** [comment]
Hi @hminle,

This is, for practical purposes the same and should give you the same result, caveating numerical differences because the former uses the 4 decimal places rounded RGB to XYZ matrix, while the latter uses the primaries and whitepoint to regenerate a RGB to XYZ matrix at full precision.

Cheers,

Thomas

**Reply by hminle** [reply]
Hi @KelSolaar ,

I got your points.
I've just done some experiments to see the effects.
![image](https://user-images.githubusercontent.com/16201681/124310747-a7848c80-db3a-11eb-8f94-1245a443a47a.png)

2 methods are the same within 1e-4 precision. 
And we should not let `chromatic_adaptation_transform=None` for colourspaces with different illuminants.


How about we convert sRGB pixels and ProPhotoRGB pixels to Lab space (use the combination of two functions `colour.RGB_to_XYZ` and `colour.XYZ_to_Lab`), and then compute the distance using `colour.algebra.euclidean_distance`?
In this case we should make sure we convert with the same illuminant, right? (similar when we want to visualize them in the same reference space https://github.com/colour-science/colour/discussions/750#discussioncomment-287937)
Like this:

```python
sRGB = np.random.rand(1,3)
prophotoRGB = np.random.rand(1,3)

ILLUMINANT_D50 = colour.CCS_ILLUMINANTS['CIE 1931 2 Degree Standard Observer']['D50']
prophotorgb_colourspace = colour.models.RGB_COLOURSPACE_PROPHOTO_RGB
srgb_colourspace_D50 = colour.models.RGB_COLOURSPACE_sRGB.chromatically_adapt(ILLUMINANT_D50)


sRGB_XYZ = colour.RGB_to_XYZ(sRGB, 
                             srgb_colourspace_D50.whitepoint, 
                             srgb_colourspace_D50.whitepoint,
                             srgb_colourspace_D50.matrix_RGB_to_XYZ)
prophotoXYZ = colour.RGB_to_XYZ(prophotoRGB,
                                prophotorgb_colourspace.whitepoint,
                                prophotorgb_colourspace.whitepoint,
                                prophotorgb_colourspace.matrix_RGB_to_XYZ)

sRGB_Lab = colour.XYZ_to_Lab(sRGB_XYZ, illuminant=ILLUMINANT_D50)
prophotoLab = colour.XYZ_to_Lab(prophotoXYZ, illuminant=ILLUMINANT_D50)
distance = colour.algebra.euclidean_distance(sRGB_Lab, prophotoLab)
```

**Reply by thomasmansencal** [reply]
> And we should not let chromatic_adaptation_transform=None for colourspaces with different illuminants.

This is actually useful, sometimes you do want to change the whitepoint of the values, let me explain:

Let's assume a display device whose whitepoint is not matching your calibration standard whitepoint, e.g. D65, but has a different colour temperature, e.g. CCT 6900K, +0.05. If you were to display imagery on that display, the colours, especially achromatic ones, would appear noticeably bluish compared to the same imagery exhibited on the D65 calibrated display. This is a case where you need to disable chromatic adaptation when generating the correction matrix: You need to transform the achromatic colours so that when they are exhibited onto that display, they end up at D65, this is where `chromatic_adaptation_transform=None` becomes critical.

> Like this:

Haven't tested the code but this should work yep, the important bit being that all the colours are compared with the same basis, i.e. the same space.
